### PR TITLE
fix: fix es-build integration compatibility issues

### PR DESCRIPTION
**Pull request by doriaviram**
- Fix es-build serverless plugin integration with lambda layers

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -97,8 +97,22 @@ class LumigoPlugin {
 		).toLowerCase();
 	}
 
+	get useServerlessEsbuild() {
```
Comment (nirLumigo) [inline src/index.js:100]: Will the new approach work also for versions before 1.30?

File: src/index.test.js [hunk 2]
```diff
@@ -528,15 +541,15 @@ describe("Lumigo plugin (python)", () => {
...
 				});
 
-				test("layers are added during after:deploy:function:initialize", async () => {
+				test("layers are added during after:package:createDeploymentArtifacts", async () => {
 					options.function = "hello";
-					await lumigo.afterDeployFunctionInitialize();
+					await lumigo.afterCreateDeploymentArtifacts();
 
 					const functions = serverless.service.functions;
 					expect(functions.hello.handler).toBe(
```
Comment (nirLumigo) [inline src/index.test.js:555]: Can you also add an integration-test? Check the file `scripts/checks.sh` and the directory `integration-test`.

**Review by nirLumigo — CHANGES_REQUESTED** [review]
Nice! :)

File: src/index.js (hunk 1 above)
Comment (doriaviram) [reply src/index.js:100]: Yes

File: src/index.test.js (hunk 2 above)
Comment (doriaviram) [reply src/index.test.js:555]: Nice !!!

File: integration-test/nodejs-esbuild/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "",
+  "main": "handler.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+      "esbuild": "^0.14.25",
+      "serverless-esbuild": "^1.25.0"
```
Comment (nirLumigo) [inline integration-test/nodejs-esbuild/package.json:13]: Don't we want at least version 1.30?

File: integration-test/nodejs-esbuild/serverless.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+service: serverless-plugin-nodejs-layers
+
+custom:
+  lumigo:
+    token: 1234
+    useLayers: true
```
Comment (nirLumigo) [inline integration-test/nodejs-esbuild/serverless.yml:6]: I think that it's suppose to work without `useLayer`, we suppose to identify `serverless-esbuild` plugin.

File: scripts/checks.sh [hunk 5]
```diff
@@ -28,6 +28,34 @@ echo
...
+pushd integration-test/nodejs-esbuild
+
+random=$RANDOM
+
+echo "** Testing NodeJS with es-build **"
+echo "********************"
+echo
+echo
+
+echo "** Deploying **"
```
Comment (nirLumigo) [inline scripts/checks.sh:41]: We print `"** Deploying **"` twice.

**Review by nirLumigo — CHANGES_REQUESTED** [review]
Great :)

**Review by nirLumigo — APPROVED** [review]

**Pull request merged**